### PR TITLE
Adding support for DurableActivityContextBase

### PR DESCRIPTION
**Pull request by gled4er**
Hello @cgillum,

I wanted to start the discussion about adding support for `DurableActivityContextBase` as described in https://github.com/Azure/azure-functions-durable-extension/issues/414  with concrete code.

Please let me know if the approach in this PR is good one or we need a change in the course.

Thank you!

### CODE REVIEW

File: samples/VSSample.Tests/HttpStartTests.cs [hunk 1]
```diff
@@ -40,6 +41,11 @@ public async Task HttpStart_returns_retryafter_header()
                 {
                     StatusCode = HttpStatusCode.OK,
                     Content = new StringContent(string.Empty),
+                    Headers =
+                    {
+                        RetryAfter = 
+                        new RetryConditionHeaderValue(TimeSpan.FromSeconds(10))
```
Comment (cgillum) [inline samples/VSSample.Tests/HttpStartTests.cs]: nit: Can this go on the same line as `RetryAfter`?

File: samples/VSSample.Tests/HelloSequenceActivityTests.cs [hunk 2]
```diff
@@ -3,14 +3,18 @@
...
+    using Microsoft.Azure.WebJobs;
     using Xunit;
+    using Moq;
 
     public class HelloSequenceActivityTests
     {
         [Fact]
         public void SayHello_returns_greeting()
         {
-            var result = HelloSequence.SayHello("John");
```
Comment (cgillum) [inline samples/VSSample.Tests/HelloSequenceActivityTests.cs]: Instead of replacing the existing test, can we use both variants?

**Comment by gled4er** [comment]
Hello @cgillum,

Thank you for reviewing this PR.

I will work on fixing the issues.

Thank you!

**Comment by gled4er** [comment]
Hello @cgillum,

I see some of the tests are failing.

I will look into it.

Thank you!

**Comment by gled4er** [comment]
Hello @cgillum,

I made some changes and run all the 306 tests locally and then pass for me locally. However, the CI is always failing. Do you have any idea what can be wrong?

Thank you!

**Comment by gled4er** [comment]
Hello @cgillum,

I run once more all 306 tests and they passed again without issues as well. One difference can be that I am using 16 GB VM in Azure for that. 

Let me know if you have any ideas.

Thank you!

**Comment by gled4er** [comment]
Hello @cgillum,

I think the changes made by the Purging History code for adding a service message in the Instances table in case of large input caused the failing tests. I fixed them and now I am expecting the tests to pass.

Thank you!

**Pull request merged**